### PR TITLE
restructure & add compile/release stuff

### LINKED ISSUE #2: release a compiled binary or whatever so it can be easily run in an environment without deno installed
main use case for this is being able to quickly and easily expose command line utilities from their docker images so you can call them from another container, if you need to add all the dependencies for this into the image then it defeats the purpose (like most other projects that attempt to do this)

### PR DESCRIPTION
**Pull request by DetachHead**
closes #2

### CODE REVIEW

File: .github/workflows/check-and-release.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      - uses: actions/checkout@v2
+      - uses: denoland/setup-deno@v1
+        with:
+          deno-version: ${{ env.DENO_VERSION }}
+      - run: |
+          mkdir dist
+          for target in x86_64-pc-windows-msvc x86_64-unknown-linux-gnu x86_64-apple-darwin; do
+            deno compile --no-check $DENO_ARGS --target=$target --output=dist/shell-http-$target src/main.ts
+          done
+        name: compile
```
Comment (KotlinIsland) [inline .github/workflows/check-and-release.yaml:45]: image
sus

File: .github/workflows/todo checker.yaml [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+      - uses: 'actions/checkout@master'
+      - name: 'TODO to Issue'
+        uses: 'alstr/todo-to-issue-action@master'
+        with:
+          REPO: ${{ github.repository }}
+          BEFORE: ${{ github.event.before }}
+          SHA: ${{ github.sha }}
+          TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          LABEL: 'TODO'
+        id: 'todo'
```
Comment (KotlinIsland) [inline .github/workflows/todo checker.yaml:18]: 4 spaces? other files have 2

File: .github/workflows/check-and-release.yaml (hunk 1 above)
Comment (DetachHead) [reply .github/workflows/check-and-release.yaml:45]: image
yea looks like a defect in github's syntax highlighting

File: .github/workflows/todo checker.yaml (hunk 2 above)
Comment (DetachHead) [reply .github/workflows/todo checker.yaml:18]: this is 2

Comment (KotlinIsland) [reply .github/workflows/todo checker.yaml:18]: 🤣

**Comment by KotlinIsland** [comment]
LGTM!

**Pull request merged**